src/domain/enrolment/utils.ts

```diff
@@ -39,6 +40,32 @@ export const getEnrolmentPayload = ({
   };
 };
 
+export const getQueuePayload = ({
+  pEventId,
+  values,
+}: {
+  pEventId: string;
+  values: EnrolmentFormFields;
+}): EnrolEventQueueMutationInput => {
+  return {
+    pEventId,
+    notificationType: getNotificationType(values),
+    person: values.isSameResponsiblePerson
+      ? undefined
+      : { ...values.person, language: values.language as Language },
+    studyGroup: {
+      ...values.studyGroup,
+      amountOfAdult: Number(values.studyGroup.amountOfAdult),
+      groupSize: Number(values.studyGroup.groupSize),
+      person: {
+        ...values.studyGroup.person,
+        language: values.language as Language,
+      },
+      studyLevels: values.studyGroup.studyLevels,
+    },
+  };
```

Comment:
I would take the getEnrolmentPayload as a base and override the base from where it is needed, meaning the ids. Would it make sense to have a new function to get the base for both, the `getEnrolmentPayload` and the new `getQueuePayload`?

For example 

```typescript
export const getEnrolmentPayloadBase = ({
  values,
}: {
  values: EnrolmentFormFields;
}): EnrolOccurrenceMutationInput => {
  return {
    notificationType: getNotificationType(values),
    person: values.isSameResponsiblePerson
      ? undefined
      : { ...values.person, language: values.language as Language },
    studyGroup: {
      ...values.studyGroup,
      amountOfAdult: Number(values.studyGroup.amountOfAdult),
      groupSize: Number(values.studyGroup.groupSize),
      person: {
        ...values.studyGroup.person,
        language: values.language as Language,
      },
      studyLevels: values.studyGroup.studyLevels,
    },
  };
};
```

```typescript
export const getEnrolmentPayload = ({
  occurrenceIds,
  values,
}: {
  occurrenceIds: string[];
  values: EnrolmentFormFields;
}): EnrolOccurrenceMutationInput => {
  return {
    ...getEnrolmentPayloadBase(values),
    occurrenceIds,
  };
};
```

```typescript
export const getQueuePayload = ({
  pEventId,
  values,
}: {
  pEventId: string;
  values: EnrolmentFormFields;
}): EnrolEventQueueMutationInput => {
  return {
    ...getEnrolmentPayloadBase(values),
    pEventId,
    },
  };
};
```

Reply:
Like this it would be clear that they share a lot of similarities. And the whole point is to do enrolments out of the queue-enrolments when there is some space left in the occurrence, so after all, they really should always share as match similarities as possible.

Reply:
yes, agree